src/DotLiquid/Tags/For.cs

```diff
@@ -200,6 +200,9 @@ public override void Render(Context context, TextWriter result)
 
         private static List<object> SliceCollectionUsingEach(Context context, IEnumerable collection, int from, int? to)
         {
+            if (collection is string @string && @string.Length > 0)
+                return new List<object> { @string };

```

Comment:
I agree that this will make DotLiquid consistent with Ruby implementation. However, if a user of DotLiquid was transforming a string character by character through the for loop, this change will break them, right?

Reply:
Yes, it will. I didn't imagine you would allow this, but I'm just prepping for the inevitable 3.0 branch to break free of these constraints.

Reply:
Ok. Sorry, I don't mean to be a pain in the rear, just looking out for the impact of code changes.

Reply:
@daviburg Please approve my other PR for new Liquid syntax as it also includes DotLiquid 2.4 syntax. That will go to a new devel branch where myself and @RodneyRichardson will commit other syntax changes with 2.4. Once done, we'll merge that all into master. I'm ready to add a syntax check here since anyways context is available.